### PR TITLE
Fr/4 add csr instruction and register support

### PR DESCRIPTION
**Pull request by deveshshevde**
csr reg dump + toolchain version independent

### CODE REVIEW

**Comment by ghost** [comment]
@deveshshevde it appears that you are not following git rebase strategy to keep your branch updated. This will always end up in conflicts.

Close this PR and raise a new one based on "rebase" strategy!

File: Makefile [hunk 1]
```diff
@@ -28,6 +32,12 @@ INC_FLAGS := $(addprefix -I,$(INC_DIRS))
 # These files will have .d instead of .o as the output.
 CPPFLAGS := $(INC_FLAGS) -MMD -MP
 
+ifeq ($(shell test $(GCC_MAJOR) -ge 11 && echo 1), 1)
+    MARCH_FLAG := -march=rv32i_zicsr
+else
+    MARCH_FLAG := -march=rv32i
+endif
```
Comment (ghost) [inline Makefile:34]: not a good way to proceed. use static assignment march_flag, usually they are backward compatible.

File: src/common/main.c [hunk 2]
```diff
@@ -118,6 +118,11 @@ static void rv32_cpu_reg_dump(char const *asm_name)
         sprintf(line, "%4s: 0x%08x\n", reg_name_list[i], *((uint32_t *)g_rv32i_ctx + i));
         fputs(line, mem);
     }
+    for (size_t i = 0; i < sizeof(rv32i_csr_t) / sizeof(uint32_t); i++)
+    {
+        sprintf(line, "%s: 0x%08x\n", csr_reg_list[i], *((uint32_t* )&g_rv32i_csr + i));
+        fputs(line, mem);
+    }
```
Comment (ghost) [inline src/common/main.c]: @pranjalchanda08 @deveshshevde how do you plan to 64b support?

**Review by ghost — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (deveshshevde) [reply Makefile:34]: Oh ok will do

**Comment by ghost** [comment]
> @deveshshevde it appears that you are not following git rebase strategy to keep your branch updated. This will always end up in conflicts.
> 
> Close this PR and raise a new one based on "rebase" strategy!

Highlighting this ...

**Comment by pranjalchanda08** [comment]
Hi @deveshshevde Please solve the issue.

File: Makefile (hunk 1 above)
Comment (pranjalchanda08) [reply Makefile:34]: The flag does not work for older compilers. We can proceed with this way for now

File: src/common/main.c (hunk 2 above)
Comment (pranjalchanda08) [reply src/common/main.c]: Currently we are focusing on rv32 only but shall revisit all the requirements and changes for RV64/128...

**Pull request closed without merging**